### PR TITLE
Fix failing tests, Fix #347 and Fix #346

### LINKED ISSUE #346: NullFieldMarshallingTest.testAbstractNullFieldUnmarshalledCorrectlyBinary is failing
Before #341 it didn't fail, on 5c81549982dd23b21e4c44f815d043cdd86f0b6b it will always pass.

### LINKED ISSUE #347: Fixed JSON Wire with ZoneID
see failing test - net.openhft.chronicle.wire.TestMarshallableZoneId

### PR DESCRIPTION
**Pull request by minborg**
The way deserialization should be done is a bit vague. This PR fixes the outstanding failing tests while trying to be as close to the old behaviour as possible.

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/wire/ReadmeChapter1Test.java [hunk 1]
```diff
@@ -181,6 +182,13 @@ public void example2() {
         bytes2.releaseLast();
     }
 
+    @Test
+    public void abstr() {
```
Comment (alamar) [inline src/test/java/net/openhft/chronicle/wire/ReadmeChapter1Test.java]: This test should have checks maybe?

**Review by alamar — CHANGES_REQUESTED** [review]
This change does not fix my failing test https://github.com/OpenHFT/Chronicle-Wire/pull/351
Perhaps it should

File: src/test/java/net/openhft/chronicle/wire/ReadmeChapter1Test.java (hunk 1 above)
Comment (minborg) [reply src/test/java/net/openhft/chronicle/wire/ReadmeChapter1Test.java]: Ahh. This test should be removed. I will do that in a new commit

**Comment by minborg** [comment]
We could fix that in a separate PR

**Review by alamar — APPROVED** [review]

**Review by glukos — APPROVED** [review]

**Pull request merged**